services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3965,6 +3969,61 @@ void setAccessControl(String clusterName, String storeName, boolean accessContro
     });
   }
 
+  @Override
+  public StoreMetaValue getMetaStoreValue(StoreMetaKey metaKey, String storeName) {
+    String metaStoreName = VeniceSystemStoreType.META_STORE.getSystemStoreName(storeName);
+    String d2Service = discoverCluster(storeName).getSecond();
+
+    try (AvroSpecificStoreClient<StoreMetaKey, StoreMetaValue> client = ClientFactory.getAndStartSpecificAvroClient(
+        ClientConfig.defaultSpecificClientConfig(metaStoreName, StoreMetaValue.class)
+            .setD2ServiceName(d2Service)
+            .setD2Client(this.d2Client))) {
+      return client.get(metaKey).get();
+    } catch (Exception e) {
+      LOGGER.warn("Could not fetch meta store value for key " + metaKey, e);
+      return null;
+    }
+  }
+
+  @Override
+  public Set<Integer> getInUseValueSchemaIds(String clusterName, String storeName) {
+    Store store = getStore(clusterName, storeName);
+    Set<Integer> schemaIds = new HashSet<>();
+
+    // Fetch value schema id used by all existing store version
+    for (Version version: store.getVersions()) {
+      Map<String, String> map = new HashMap<>(2);
+      map.put(MetaStoreWriter.KEY_STRING_VERSION_NUMBER, Integer.toString(version.getNumber()));
+      StoreMetaKey key = MetaStoreDataType.VALUE_SCHEMAS_WRITTEN_PER_STORE_VERSION.getStoreMetaKey(map);
+      StoreMetaValue metaValue = getMetaStoreValue(key, storeName);
+
+      if (metaValue == null) {
```

Comment:
`null` means that the meta store is not working properly, right?
Shall we throw an exception instead?

Reply:
also it could be just a newly populated version which has no writes yet ?

Reply:
Sure, either way, I think we should throw exception to make sure we don't delete the schema, which can be potentially used soon.